packages/liveblocks-yjs/src/doc.ts

```diff
@@ -38,19 +45,42 @@ export default class yDocHandler extends Observable<unknown> {
     };
 
     this.syncDoc();
+
+    this.localSnapshotΣ = new Signal<Y.Snapshot>(Y.snapshot(doc));
+    this.remoteSnapshotΣ = new Signal<Y.Snapshot | null>(null);
+
+    this.syncStatusSignalΣ = DerivedSignal.from(
+      this.localSnapshotΣ,
+      this.remoteSnapshotΣ,
+      (localSnapshot, remoteSnapshot) => {
+        if (remoteSnapshot === null) {
+          return "loading";
+        }
+        const isEqual = Y.equalSnapshots(localSnapshot, remoteSnapshot);
+        return isEqual ? "synchronized" : "synchronizing";
+      }
+    );
   }
 
   public handleServerUpdate = ({
     update,
     stateVector,
     readOnly,
     v2,
+    remoteSnapshot: _remoteSnapshot,
   }: {
     update: Uint8Array;
     stateVector: string | null;
     readOnly: boolean;
     v2?: boolean;
+    remoteSnapshot: Uint8Array;
   }): void => {
+    this.remoteSnapshotΣ.set(
+      v2
+        ? Y.decodeSnapshotV2(_remoteSnapshot)
+        : Y.decodeSnapshot(_remoteSnapshot)
+    );
```

Comment:
When a remote update is received, we update the remote snapshot value. Similarly, when a  local update is made, we update the local snapshot value. When either value changes, we calculate the sync status and store it in the `syncStatusSignalΣ` signal. 


```ts
this.syncStatusSignalΣ = DerivedSignal.from(
  this.localSnapshotΣ,
  this.remoteSnapshotΣ,
  (localSnapshot, remoteSnapshot) => {
    if (remoteSnapshot === null) {
      return "loading";
    }
    const isEqual = Y.equalSnapshots(localSnapshot, remoteSnapshot);
    return isEqual ? "synchronized" : "synchronizing";
  }
);
```

Reply:
I'm worried about the performance of this on old documents.

This will potentially run per key-stroke, and users are potentially sensitive to typing delays. 

I'm wondering if maybe a better approach is to debounce this.